src/canvas/Curve.tsx

```diff
@@ -22,29 +29,21 @@ export class Curve {
 
     getLength = (): number => {
         let length = 0;
-        // compute the total distance from start to control + from control to end in order to compute an 
-        // appropriate number of points on the curve.
-        const NUMPOINTS = Math.ceil((Math.sqrt(this.start.distanceSquared(this.control)) + Math.sqrt(this.control.distanceSquared(this.end))) / 5);
+        // compute the total distance from start to control + from control to end in order to 
+        // find an upper bound on the curve length.
+        const startToControl = this.start.distanceTo(this.control);
+        const controlToEnd = this.control.distanceTo(this.end);
+        const upperBound = Math.ceil(startToControl + controlToEnd);
+        // find the number of points to take on the curve as a function of the upperbound
+        const NUMPOINTS = upperBound / 5;
         const pointsOnCurve = this.computePointsOnCurve(NUMPOINTS);
         for (let i = 0; i < NUMPOINTS - 1; i++) {
-            length += Math.sqrt(pointsOnCurve[i+1].distanceSquared(pointsOnCurve[i]));
+            length += pointsOnCurve[i+1].distanceTo(pointsOnCurve[i]);
         }
         return length;
     };
 
-    // using De Casteljau's algorithm (slower but more stable than the direct approach)
-    private computePoint = (t: number): Point => {
-        const startToControlX = this.lerp(this.start.getX(), this.control.getX(), t);
-        const startToControlY = this.lerp(this.start.getY(), this.control.getY(), t);
-
-        const controlToEndX = this.lerp(this.control.getX(), this.end.getX(), t);
-        const controlToEndY = this.lerp(this.control.getY(), this.end.getY(), t);
-
-        return new Point(this.lerp(startToControlX, controlToEndX, t),
-                    this.lerp(startToControlY, controlToEndY, t));
-    };
-
-    private lerp = (start: number, end: number, t: number): number => {
+    protected lerp = (start: number, end: number, t: number): number => {
         return start + t * (end - start);
     };
```

Comment:
Do you think we should have an abstract shape class that has abstract functions getLength and draw, which Curve would Implement. I also wonder if we should have a line class for consistency, but I am not sure if this is over engineering it.

If not, maybe we should have an  'ICurve' type to the types file that Curve will implement?

Reply:
Yeah I thought about the line too, and also thought it might be overkill for now. But I now realize probably going to add more abstract functions to "Shape" for adding seam allowance, so maybe it would be worth it

Reply:
I'm not sure I like the name "shape" though. It sounds like it would refer to circles, squares, triangles... How about Stroke?

Reply:
Per our conversation on 8/13 we decided to go with Segment as the abstract class that other classes, such as  Line, Curve, inherit from.